### ISSUE TITLE
Failed to fork child process

### ISSUE THREAD

**Issue by Patriot2407** [husisusi/officeonlin-install.sh#140]
I have tried running your script on many different machines all with Ubuntu 16.04 LTS installed. No matter what I do, it always fails to start. Same exact error on all machines.

**Comment by Patriot2407** [comment]
[ forkit ] FTL  Neither /opt/libreoffice/instdir/program/libsofficeapp.so or /opt/libreoffice/instdir/program/libmergedlo.so existwsd-04117-04117 21:11:52.808442 [ loolwsd ] FTL  Failed to fork child processes.| wsd/LOOLWSD.cpp:2521
Failed to fork child processes.
wsd-04117-04117 21:11:52.808661 [ loolwsd ] FTL  Failed to fork child processes.| wsd/LOOLWSD.cpp:2671
Failed to fork child processes.
wsd-04117-04117 21:11:52.808910 [ loolwsd ] WRN  Waking up dead poll thread [admin], started: false, finished: false| ./net/Socket.hpp:507
wsd-04117-04117 21:11:52.809002 [ loolwsd ] WRN  Waking up dead poll thread [admin], started: false, finished: false| ./net/Socket.hpp:507
<shutdown>-04117 21:11:52.809532 [ loolwsd ] WRN  Waking up dead poll thread [accept_poll], started: false, finished: false| ./net/Socket.hpp:507
<shutdown>-04117 21:11:52.809563 [ loolwsd ] WRN  Waking up dead poll thread [accept_poll], started: false, finished: false| ./net/Socket.hpp:507
<shutdown>-04117 21:11:52.809576 [ loolwsd ] WRN  Waking up dead poll thread [websrv_poll], started: false, finished: false| ./net/Socket.hpp:507
<shutdown>-04117 21:11:52.809588 [ loolwsd ] WRN  Waking up dead poll thread [websrv_poll], started: false, finished: false| ./net/Socket.hpp:507
<shutdown>-04117 21:11:52.809599 [ loolwsd ] WRN  Waking up dead poll thread [accept_poll], started: false, finished: false| ./net/Socket.hpp:507
<shutdown>-04117 21:11:52.809609 [ loolwsd ] WRN  Waking up dead poll thread [accept_poll], started: false, finished: false| ./net/Socket.hpp:507
<shutdown>-04117 21:11:52.809909 [ loolwsd ] WRN  Waking up dead poll thread [prisoner_poll], started: true, finished: true| ./net/Socket.hpp:507
<shutdown>-04117 21:11:52.809965 [ loolwsd ] WRN  Waking up dead poll thread [prisoner_poll], started: true, finished: true| ./net/Socket.hpp:507
<shutdown>-04117 21:11:52.810036 [ loolwsd ] WRN  Waking up dead poll thread [websrv_poll], started: false, finished: false| ./net/Socket.hpp:507
<shutdown>-04117 21:11:52.810074 [ loolwsd ] WRN  Waking up dead poll thread [websrv_poll], started: false, finished: false| ./net/Socket.hpp:507

**Comment by Patriot2407** [comment]
Everything successfully compiles, it seems, and i get the yellow box saying that loolwsd.service has started. It does say it starts, but after 1 minute it gives up and I check the status and it gives the above error.

**Comment by aalaesar** [comment]
Seeing Neither /opt/libreoffice/instdir/program/libsofficeapp.so or /opt/libreoffice/instdir/program/libmergedlo.so exist
Suggest the libre office compilation maybe bad
Did you used all default parameters ?

**Comment by aalaesar** [comment]
Also the final test kill the service after 15 seconds. And the kill generate similar errors. (Because the test service is unsafe)

It has to be started  with systemctl after that.
I suppose you did that.:relaxed:

**Comment by Patriot2407** [comment]
Yes I did start with systemctl. Okay thanks, I will attempt a recompile on libre office. Yes I used the default paramters for compile.

**Comment by Patriot2407** [comment]
You are correct about Libre Office not being compiled completely. I just went back through the logs and it never successfully compiled. ANd i am forcing a recompile, but to no avail.

**Comment by Patriot2407** [comment]
Build always fails right here
[build CXX] vcl/headless/svptext.cxx
/opt/libreoffice/vcl/null/printerinfomanager.cxx: In constructor â€˜psp::PrinterInfoManager::PrinterInfoManager(psp::PrinterInfoManager::Type)â€™:
/opt/libreoffice/vcl/null/printerinfomanager.cxx:47:33: error: call of overloaded â€˜unique_ptr(NULL)â€™ is ambiguous
     m_aSystemDefaultPaper( "A4" )
                                 ^
In file included from /usr/include/c++/5/memory:81:0,
                 from /opt/libreoffice/include/i18nlangtag/languagetag.hxx:19,
                 from /opt/libreoffice/include/tools/resmgr.hxx:23,
                 from /opt/libreoffice/include/tools/rc.hxx:23,
                 from /opt/libreoffice/include/vcl/outdev.hxx:26,
                 from /opt/libreoffice/include/vcl/window.hxx:25,
                 from /opt/libreoffice/include/vcl/vclevent.hxx:26,
                 from /opt/libreoffice/include/vcl/helper.hxx:26,
                 from /opt/libreoffice/include/vcl/printerinfomanager.hxx:27,
                 from /opt/libreoffice/vcl/null/printerinfomanager.cxx:20:
/usr/include/c++/5/bits/unique_ptr.h:356:7: note: candidate: std::unique_ptr<_Tp, _Dp>::unique_ptr(const std::unique_ptr<_Tp, _Dp>&) [with _Tp = psp::SystemQueueInfo; _Dp = std::default_delete<psp::SystemQueueInfo>] <deleted>
       unique_ptr(const unique_ptr&) = delete;
       ^
In file included from /usr/include/c++/5/memory:81:0,
                 from /opt/libreoffice/include/i18nlangtag/languagetag.hxx:19,
                 from /opt/libreoffice/include/tools/resmgr.hxx:23,
                 from /opt/libreoffice/include/tools/rc.hxx:23,
                 from /opt/libreoffice/include/vcl/outdev.hxx:26,
                 from /opt/libreoffice/include/vcl/window.hxx:25,
                 from /opt/libreoffice/include/vcl/vclevent.hxx:26,
                 from /opt/libreoffice/include/vcl/helper.hxx:26,
                 from /opt/libreoffice/include/vcl/printerinfomanager.hxx:27,
                 from /opt/libreoffice/vcl/null/printerinfomanager.cxx:20:
/usr/include/c++/5/bits/unique_ptr.h:205:7: note: candidate: std::unique_ptr<_Tp, _Dp>::unique_ptr(std::unique_ptr<_Tp, _Dp>&&) [with _Tp = psp::SystemQueueInfo; _Dp = std::default_delete<psp::SystemQueueInfo>]
       unique_ptr(unique_ptr&& __u) noexcept
       ^
/usr/include/c++/5/bits/unique_ptr.h:200:17: note: candidate: constexpr std::unique_ptr<_Tp, _Dp>::unique_ptr(std::nullptr_t) [with _Tp = psp::SystemQueueInfo; _Dp = std::default_delete<psp::SystemQueueInfo>; std::nullptr_t = std::nullptr_t]
       constexpr unique_ptr(nullptr_t) noexcept : unique_ptr() { }
                 ^
In file included from /usr/include/c++/5/memory:81:0,
                 from /opt/libreoffice/include/i18nlangtag/languagetag.hxx:19,
                 from /opt/libreoffice/include/tools/resmgr.hxx:23,
                 from /opt/libreoffice/include/tools/rc.hxx:23,
                 from /opt/libreoffice/include/vcl/outdev.hxx:26,
                 from /opt/libreoffice/include/vcl/window.hxx:25,
                 from /opt/libreoffice/include/vcl/vclevent.hxx:26,
                 from /opt/libreoffice/include/vcl/helper.hxx:26,
                 from /opt/libreoffice/include/vcl/printerinfomanager.hxx:27,
                 from /opt/libreoffice/vcl/null/printerinfomanager.cxx:20:
/usr/include/c++/5/bits/unique_ptr.h:169:7: note: candidate: std::unique_ptr<_Tp, _Dp>::unique_ptr(std::unique_ptr<_Tp, _Dp>::pointer) [with _Tp = psp::SystemQueueInfo; _Dp = std::default_delete<psp::SystemQueueInfo>; std::unique_ptr<_Tp, _Dp>::pointer = psp::SystemQueueInfo*]
       unique_ptr(pointer __p) noexcept
       ^
In file included from /usr/include/c++/5/memory:81:0,
                 from /opt/libreoffice/include/i18nlangtag/languagetag.hxx:19,
                 from /opt/libreoffice/include/tools/resmgr.hxx:23,
                 from /opt/libreoffice/include/tools/rc.hxx:23,
                 from /opt/libreoffice/include/vcl/outdev.hxx:26,
                 from /opt/libreoffice/include/vcl/window.hxx:25,
                 from /opt/libreoffice/include/vcl/vclevent.hxx:26,
                 from /opt/libreoffice/include/vcl/helper.hxx:26,
                 from /opt/libreoffice/include/vcl/printerinfomanager.hxx:27,
                 from /opt/libreoffice/vcl/null/printerinfomanager.cxx:20:
/usr/include/c++/5/bits/unique_ptr.h: In instantiation of â€˜void std::default_delete<_Tp>::operator()(_Tp*) const [with _Tp = psp::SystemQueueInfo]â€™:
/usr/include/c++/5/bits/unique_ptr.h:236:17:   required from â€˜std::unique_ptr<_Tp, _Dp>::~unique_ptr() [with _Tp = psp::SystemQueueInfo; _Dp = std::default_delete<psp::SystemQueueInfo>]â€™
/opt/libreoffice/vcl/null/printerinfomanager.cxx:47:33:   required from here
/usr/include/c++/5/bits/unique_ptr.h:74:22: error: invalid application of â€˜sizeofâ€™ to incomplete type â€˜psp::SystemQueueInfoâ€™
  static_assert(sizeof(_Tp)>0,
                      ^
make[1]: *** [/opt/libreoffice/workdir/CxxObject/vcl/null/printerinfomanager.o] Error 1
/opt/libreoffice/solenv/gbuild/LinkTarget.mk:189: recipe for target '/opt/libreoffice/workdir/CxxObject/vcl/null/printerinfomanager.o' failed
make[1]: *** Waiting for unfinished jobs....
make: *** [build] Error 2
Makefile:263: recipe for target 'build' failed

**Comment by Patriot2407** [comment]
Can I try a version that we know works? Does anyone have somwehat current build that can point me in the right direction?

**Comment by Patriot2407** [comment]
Failing at the exact same spots on a fresh VM of Ubuntu 16.04. Fails to build libreoffice.

**Comment by Patriot2407** [comment]
I ran script with all defaults. Like this:
git clone https://github.com/husisusi/officeonlin-install.sh.git
cd office*
chmod +x office*.sh
./office*.sh

**Comment by aalaesar** [comment]
This looks like a issue between the core project components and a version of the vcl library on the system.
I would like to know what branches and tags the scrip has selected for the build. They should be shown at the beginning  of the logs

**Comment by aalaesar** [comment]
You can use options to force the script to build from a specific  tag or commit. More infos in the read me and in the config file itself. 
Regards
 Aal.

**Comment by Patriot2407** [comment]
Switched to a new branch 'distro/collabora/cp-5.3'
Branch distro/collabora/cp-5.3 set up to track remote branch distro/collabora/cp-5.3 from origin.
Selected Tag: cp-5.3-33 a1d80cc
HEAD is now at a1d80cc Bump version to 5.3-33

**Comment by Patriot2407** [comment]
Doesn't lool depend on the version of libreoffice though? So if I change the tag or commit, won't I need to change it for lool as well?

**Comment by Patriot2407** [comment]
Alright, I am trying libreoffice cp-5.3-32 with Change-Id: Ia81627cb441158edd1e24d4cfb4b9c3f95853ec6. I will report back if this works, once it is done.

**Comment by Kassiematis** [comment]
Sometimes it helps to remove /opt/libreoffice and /home/lool/.ccache before compile.

**Comment by Patriot2407** [comment]
Alrght. I will try this. Like I said, I did it on a fresh install of Ubuntu 16.04 LTS and it still did not work. As of now, it still fails to compile even on libreoffice cp-5.3-32 with Change-Id: Ia81627cb441158edd1e24d4cfb4b9c3f95853ec6

**Comment by Kassiematis** [comment]
Try changing lib/default.cfg line 19 from

declare -r lo_mini_opts='--without-help --without-myspell-dicts --without-java --without-doxygen
--disable-cups --disable-dconf --disable-odk --without-junit --disable-dbus --disable-firebird-sdbc
--disable-postgresql-sdbc --disable-gltf --disable-extensions --disable-pdfimport --disable-neon
--disable-lpsolve --disable-coinmp --disable-systray --disable-randr --disable-gstreamer-1-0
--without-helppack-integration --disable-report-builder --disable-coinmp --disable-collada

To 

declare -r lo_mini_opts='--without-help --without-myspell-dicts'

**Comment by aalaesar** [comment]
Hi @czebiniak 
Any news ? did removing those options worked ?

For infos: issue reproduced on my side.
Investigating... 🔍

**Comment by aalaesar** [comment]
@czebiniak 
@Kassiematis is correct, this is a build option issue.
I've just built libreoffice with a set of very different options for testing and it passed compilation. So i'll look for the build option (combinaison maybe) that break the build

**Comment by Kassiematis** [comment]
Looks like the compile has a print support error.

`/opt/libreoffice/vcl/null/printerinfomanager.cxx: In constructor â€˜psp::PrinterInfoManager::PrinterInfoManager(psp::PrinterInfoManager::Type)â€™:
/opt/libreoffice/vcl/null/printerinfomanager.cxx:47:33: error: call of overloaded â€˜unique_ptr(NULL)â€™ is ambiguous
m_aSystemDefaultPaper( "A4" )`

this option could be the culprit

--disable-cups

**Comment by Patriot2407** [comment]
Hi, sorry, I got caught up in another project. I am trying the compile with removing --disable-cups to see if it solves the issue. If it does not, I will try declare -r lo_mini_opts='--without-help --without-myspell-dicts'

Thanks so much for the helpful tips!

**Comment by aalaesar** [comment]
> this option could be the culprit
--disable-cups

Confirmed on my side:
removing disable-cups removes the error.
Thanks @Kassiematis.

**Comment by Patriot2407** [comment]
I'm still getting build errors.
[build CUT] dbaccess_empty_stdlib_save
PuTTY X11 proxy: Unsupported authorisation protocol
PuTTY X11 proxy: Unsupported authorisation protocol
/opt/libreoffice/cppcanvas/qa/extras/emfplus/emfplus.cxx:91:Test::testFdo77229
equality assertion failed
- Expected: 65024
- Actual  : 16777215

Test::testFdo77229 finished in: 4126ms
emfplus.cxx:91:Assertion
Test name: Test::testFdo77229
equality assertion failed
- Expected: 65024
- Actual  : 16777215

Failures !!!
Run: 1   Failure total: 1   Failures: 1   Errors: 0

Error: a unit test failed, please do one of:
make CppunitTest_cppcanvas_emfplus CPPUNITTRACE="gdb --args"
    # for interactive debugging on Linux
make CppunitTest_cppcanvas_emfplus VALGRIND=memcheck
    # for memory checking
make CppunitTest_cppcanvas_emfplus DEBUGCPPUNIT=TRUE
    # for exception catching

/opt/libreoffice/solenv/gbuild/CppunitTest.mk:101: recipe for target '/opt/libreoffice/workdir/CppunitTest/cppcanvas_emfplus.test' failed
make[1]: *** [/opt/libreoffice/workdir/CppunitTest/cppcanvas_emfplus.test] Error 1
make[1]: *** Waiting for unfinished jobs....
Makefile:263: recipe for target 'build' failed
make: *** [build] Error 2

**Comment by Patriot2407** [comment]
Setup a new Ubuntu 16.04 LTS machine, and it still fails. Not sure why this was closed since I'm still having build issues? I never said it worked?

onfig.status: executing libtool commands
config.status: executing depfiles commands

Configuration:
    LOKit path              /opt/online/bundled/include
    LO path                 /opt/libreoffice/instdir
    LO integration tests    test against /opt/libreoffice/instdir
    SSL support             ssl enabled
    Debug & low security    secure mode: product build

    $ make # to compile
    $ make run # to start loolwsd

make: Entering directory '/opt/online'
make  all-recursive
make[1]: Entering directory '/opt/online'
Making all in .
make[2]: Entering directory '/opt/online'
  CXX      wsd/Admin.o
  CXX      wsd/AdminModel.o
  CXX      wsd/Auth.o
  CXX      wsd/DocumentBroker.o
  CXX      wsd/LOOLWSD.o
  CXX      wsd/ClientSession.o
In file included from wsd/LOOLWSD.cpp:106:0:
./common/UnitHTTP.hpp:86:18: error: â€˜virtual bool UnitHTTPServerRequest::expectContinue() constâ€™ marked â€˜overrideâ€™, but does not override
     virtual bool expectContinue() const override
                  ^
  CXX      wsd/FileServer.o
make[2]: *** [wsd/LOOLWSD.o] Error 1
make[2]: *** Waiting for unfinished jobs....
Makefile:1167: recipe for target 'wsd/LOOLWSD.o' failed
make[2]: Leaving directory '/opt/online'
make[1]: *** [all-recursive] Error 1
Makefile:1703: recipe for target 'all-recursive' failed
make[1]: Leaving directory '/opt/online'
make: *** [all] Error 2
Makefile:736: recipe for target 'all' failed
make: Leaving directory '/opt/online'

**Comment by Patriot2407** [comment]
I ran this after your latest commit that was pushed today, with the default build file modified. Yes, it successfully builds LibreOffice, but LibreOffice Online fails to build.

**Comment by aalaesar** [comment]
Hi @czebiniak 
> Not sure why this was closed since I'm still having build issues? I never said it worked?

Well, as Github issues are bug oriented, the system closed this one because the initial bug has been fixed by #142.
I understand you're expecting user support and, as you're hitting a new bug/issue, I kindly suggest you to open a new issue. ☺️ 

Regards
Aal.

**Comment by Patriot2407** [comment]
It's okay, I fixed it anyway. Looks like the latest commit to LOOL caused an issue, so I went with commit 44600de7c7dbe11da0545cf73ccae9dfa653d141,and it worked.
